test/test-mongocrypt-ctx-encrypt.c

```diff
@@ -2551,8 +2606,24 @@ _test_encrypt_fle2_unindexed_encrypted_payload (_mongocrypt_tester_t *tester)
    uint8_t rng_data[] = RNG_DATA;
    _test_rng_data_source source = {
       .buf = {.data = rng_data, .len = sizeof (rng_data) - 1u}};
-   _test_encrypt_fle2_encryption_placeholder (
-      tester, "fle2-insert-unindexed", &source);
+
+   // TODO (MONGOCRYPT-551): Implement UnindexedEncryptedValueV2 transform
+   // Add kFLE2v2Enable case (or use macro) when the above work complete.
+   source.pos = 0;
+   _test_encrypt_fle2_encryption_placeholder (tester,
+                                              "fle2-insert-unindexed",
+                                              &source,
+                                              kFLE2v2Default,
+                                              "encrypted-payload.json",
+                                              NULL);
+
+   source.pos = 0;
+   _test_encrypt_fle2_encryption_placeholder (tester,
+                                              "fle2-insert-unindexed",
+                                              &source,
+                                              kFLE2v2Disable,
+                                              "encrypted-payload.json",
+                                              NULL);
```

Comment:
```suggestion
   // TODO (MONGOCRYPT-551): Implement UnindexedEqualityValueV2 transform
   TEST_ENCRYPT_FLE2_ENCRYPTION_PLACEHOLDER (
      tester,
      "fle2-insert-unindexed",
      &source,
      "FLE2UnindexedEncryptedValueV2 not implemented")
```

Suggest making an explicit error in `_mongocrypt_fle2_placeholder_to_FLE2UnindexedEncryptedValue` for consistency with other tests:

```c
   if (kb->crypt->opts.use_fle2_v2) {
      // TODO (MONGOCRYPT-551): UnindexedEncryptedValueV2
      CLIENT_ERR ("FLE2UnindexedEncryptedValueV2 not implemented");
   }
```

Reply:
I like it. Done.